addons/onboarding/models/onboarding_step.py

```diff
@@ -60,6 +62,31 @@ def _compute_current_progress(self):
                 step.current_progress_step_id = False
                 step.current_step_state = 'not_done'
 
+    @api.constrains('onboarding_ids')
+    def check_step_on_onboarding_has_action(self):
+        for step in self:
+            if step.onboarding_ids and not step.panel_step_open_action_name:
+                raise ValidationError(_('An action name to open is required for steps linked to an onboarding panel.'))
+
+    def write(self, vals):
+        new_is_per_company = vals.get('is_per_company')
+        steps_changing_is_per_company = (
```

Comment:
I would prefer having a real "is_company_updated" variable and some "if" instead of a void recordset that nullifies some update operations, but that's more personal taste.